idaes/generic_models/properties/core/pure/tests/test_Perrys.py

```diff
@@ -227,3 +229,87 @@ def test_dens_mol_liq_comp_eqn_2(frame):
     assert value(expr) == pytest.approx(7.48194e10, rel=1e-4)
 
     assert_units_equivalent(expr, pyunits.mol/pyunits.m**3)
+
+
+@pytest.fixture()
+def frame_deprecated():
+    m = ConcreteModel()
+
+    # Create a dummy parameter block
+    m.params = Block()
+
+    m.params.config = ConfigBlock(implicit=True)
+    m.params.config.parameter_data = {
+        "dens_mol_liq_comp_coeff": {'1': 5.459,
+                                    '2': 0.30542,
+                                    '3': 647.13,
+                                    '4': 0.081},
+        "cp_mol_liq_comp_coeff": {'1': 2.7637e+05,
+                                  '2': -2.0901e+03,
+                                  '3': 8.1250e+00,
+                                  '4': -1.4116e-2,
+                                  '5': 9.3701e-06},
+        "enth_mol_form_liq_comp_ref": -285.83e3,
+        "entr_mol_form_liq_comp_ref": 69.95}
+    m.params.config.include_enthalpy_of_formation = True
+
+    # Also need to dummy configblock on the model for the test
+    m.config = ConfigBlock(implicit=True)
+    m.config.include_enthalpy_of_formation = True
+
+    m.meta_object = PropertyClassMetadata()
+    m.meta_object.default_units["temperature"] = pyunits.K
+    m.meta_object.default_units["mass"] = pyunits.kg
+    m.meta_object.default_units["length"] = pyunits.m
+    m.meta_object.default_units["time"] = pyunits.s
+    m.meta_object.default_units["amount"] = pyunits.mol
+
+    def get_metadata(self):
+        return m.meta_object
+    m.get_metadata = types.MethodType(get_metadata, m)
+    m.params.get_metadata = types.MethodType(get_metadata, m.params)
+
+    # Add necessary parameters to parameter block
+    m.params.temperature_ref = Var(initialize=273.16, units=pyunits.K)
+
+    # Create a dummy state block
+    m.props = Block([1])
+    add_object_reference(m.props[1], "params", m.params)
+
+    m.props[1].temperature = Var(initialize=273.16, units=pyunits.K)
+
+    return m
+
+
+@pytest.mark.unit
+def test_dens_mol_liq_comp_deprecated(frame_deprecated):
+    dens_mol_liq_comp.build_parameters(frame_deprecated.params)
+
```

Comment:
You should also add a check here for the deprecation message to make sure it is being recorded. To do this, you can (I think):

1. Add `caplog` as an argument after the fixture in the test declaration
2. Then check the contents of `caplog.txt` to make sure the correct message is present.

See https://docs.pytest.org/en/6.2.x/logging.html

Reply:
I'll match the structure to existing caplog checks in other test files, thanks for the suggestion @andrewlee94